provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinBuildScriptCompiler.kt

```diff
@@ -210,27 +212,41 @@ class KotlinBuildScriptCompiler(
 
     private
     fun compileBuildscriptBlock(buildscriptRange: IntRange, scriptTemplate: KClass<*>) =
-        kotlinCompiler.compileBuildscriptBlockOf(
-            scriptTemplate,
-            scriptPath,
-            script.linePreservingSubstring(buildscriptRange),
-            buildscriptBlockCompilationClassPath)
+        withKotlinCompiler {
+            compileBuildscriptBlockOf(
+                scriptTemplate,
+                scriptPath,
+                script.linePreservingSubstring(buildscriptRange),
+                buildscriptBlockCompilationClassPath)
+        }
 
     private
     fun compilePluginsBlock(pluginsRange: IntRange, scriptTemplate: KClass<*>) =
-        kotlinCompiler.compilePluginsBlockOf(
-            scriptTemplate,
-            scriptPath,
-            script.linePreservingSubstring_(pluginsRange),
-            pluginsBlockCompilationClassPath)
+        withKotlinCompiler {
+            compilePluginsBlockOf(
+                scriptTemplate,
+                scriptPath,
+                script.linePreservingSubstring_(pluginsRange),
+                pluginsBlockCompilationClassPath)
+        }
 
     private
     fun compileScriptFileFor(classPath: ClassPath) =
-        kotlinCompiler.compileGradleScript(
-            scriptTarget.scriptTemplate,
-            scriptPath,
-            script,
-            classPath)
+        withKotlinCompiler {
+            compileGradleScript(
+                scriptTarget.scriptTemplate,
+                scriptPath,
+                script,
+                classPath)
+        }
+
+    private
+    fun <T> withKotlinCompiler(action: CachingKotlinCompiler.() -> T): T =
+        try {
+            kotlinCompiler.action()
+        } catch (e: ScriptCompilationException) {
+            throw LocationAwareException(e, scriptSource, e.errors.first().lineNumber)
```

Comment:
Is a `ScriptCompilationException` guaranteed to always have at least one error?

Reply:
It is not guaranteed. Fixed

Reply:
It is.

Reply:
I explain, `compileBunchOfSources` returns `false` if `messageCollector.hasErrors()`.

Reply:
Right, thanks! I'll follow up